src/main/java/uk/nhs/hee/trainee/details/api/BasicDetailsResource.java

```diff
@@ -64,4 +70,32 @@ public ResponseEntity<PersonalDetailsDto> updateBasicDetails(
     entity = service.createProfileOrUpdateBasicDetailsByTisId(tisId, entity);
     return ResponseEntity.ok(mapper.toDto(entity));
   }
+
+  /**
+   * Update the GMC number for the authenticated trainee.
+   *
+   * @param token     The authentication token.
+   * @param gmcNumber The new GMC number.
+   * @return The updated PersonalDetails.
+   */
+  @PutMapping("/gmc-number/{gmcNumber}")
```

Comment:
Just wondering if the GMC number was something like '92/3224' (as for this doc: https://apps.tis.nhs.uk/admin/people/person/96478/edit-personal-details)  would the mapping work (i.e. is it greedy?) I will take a look locally in a minute...

Reply:
It would bail unless we encoded it client side first. It wouldn't matter too much anyway, because they'd be supplying thr GMC for NTN purposes and that requires 7 digits to be valid anyway.

Reply:
(Perhaps this is irrelevant because we only want to facilitate updates to valid 7-digit GMC numbers)